explorer/testdata/class/abstract_method_base_instance.carbon

```diff
@@ -0,0 +1,23 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+// RUN: %{explorer-run}
+// RUN: %{explorer-run-trace}
+// CHECK:STDOUT: result: 0
+
+package ExplorerTest api;
+
+abstract class A {
+    abstract fn Foo[self: Self]();
+}
+
+base class B extends A {
```

Comment:
For completeness, could we also have a test where a non-base class extends an abstract class?